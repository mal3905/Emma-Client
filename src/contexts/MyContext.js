  
import React from 'react'

export default React.createContext({
  items: [],
  categories: [],
  addCategory: () => {},
  deleteItem: () => {},
  addItem: () => {}, 
  
})
