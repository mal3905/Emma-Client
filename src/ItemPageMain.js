import React from 'react'
import Item from './Item'
import MyContext from './MyContext'
import { findItem } from './item-helpers'


export default class ItemPageMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = MyContext

  handleDeleteItem = itemid => {
    this.props.history.push(`/`)
  }

  render() {
    const { items=[] } = this.context
    const { itemid } = this.props.match.params
    const item = findItem(items, itemid) 
    return (
      <section className='ItemPageMain'>
        <Item
          itemid={item.id}
          name={item.name}
          onDeleteItem={this.handleDeleteItem}
        />
      </section>
    )
  }
}