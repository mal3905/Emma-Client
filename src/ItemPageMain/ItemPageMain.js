import React from 'react'
import Item from '../Items/Item'
import MyContext from '../contexts/MyContext'
import { findItem } from '../item-helpers'


export default class ItemPageMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = MyContext

  handleDeleteItem = itemid => {
    this.props.history.push(`/`)
  }

  render() {
    // console.log('hello') 
    let { items=[] } = this.context
    const { itemid } = this.props.match.params
    console.log(items)
    items = findItem(items, itemid)
    
    return (
      <section className='ItemPageMain'>
        <Item
          itemid={items.id}
          name={items.name}
          onDeleteItem={this.handleDeleteItem}
        />
      </section>
    )
  }
}