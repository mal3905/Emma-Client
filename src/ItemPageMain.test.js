import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ItemPageMain from './ItemPageMain'
import renderer from 'react-test-renderer';
import MyContext from './MyContext';


const context = {
    items: [{
      id: `1`,
      name: `peas`

    }]
  }

describe(`ItemPageMain component`, () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ItemPageMain />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

  it('renders the UI as expected', () => {
    const tree = renderer
        .create(<MyContext.Provider value={context}><MyContext/><ItemPageMain/></MyContext.Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
  })

  // enzyme doesn't do React context
  it.skip('renders an item  with item prop', () => {
    const props = {
      match: {
        params: {
          itemid: 'cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1'
        }
      }
    }

    const item = shallow(<ItemPageMain {...props} />, context)
      .find('Item')
    expect(toJson(item)).toMatchSnapshot()
  })

  })
