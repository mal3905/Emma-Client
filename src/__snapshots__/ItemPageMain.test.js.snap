// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemPageMain component renders a .ItemPageMain by default 1`] = `
<section
  className="ItemPageMain"
>
  <div
    className="Item"
  >
    <h2
      className="Item__title"
    >
      Dogs
    </h2>
    <button
      className="Item__delete"
      onClick={[Function]}
      type="button"
    >
       
      remove
    </button>
  </div>
</section>
`;
